From: Iustin Pop <iustin@google.com>
Date: Wed, 10 Mar 2010 10:25:31 +0000 (+0100)
Subject: Fix a python 2.6.5 compatibility
X-Git-Url: http://git.ganeti.org/?p=ganeti.git;a=commitdiff_plain;h=6a180ca8d18005ded4cbf6c13e877ebcfc84a275

Fix a python 2.6.5 compatibility

The upcoming python 2.6.5 release has a change that makes delattr(obj,
attr) fail for slots-enabled objects if the attr is not already set.

To prevent against this, we only run the delattr if the attribute is
already set.

Signed-off-by: Iustin Pop <iustin@google.com>
Reviewed-by: Guido Trotter <ultrotter@google.com>
(cherry-picked from 44db3a6f466a2cd10d3719c29985a5eb740e341d)

Signed-off-by: Iustin Pop <iustin@google.com>
Reviewed-by: Michael Hanselmann <hansmi@google.com>
---

diff --git a/lib/opcodes.py b/lib/opcodes.py
index 6a86477..f3142e1 100644
--- a/lib/opcodes.py
+++ b/lib/opcodes.py
@@ -89,7 +89,7 @@ class BaseOpCode(object):
                        type(state))
 
     for name in self.__slots__:
-      if name not in state:
+      if name not in state and hasattr(self, name):
         delattr(self, name)
 
     for name in state:
